library/ssl_cli.c

```diff
@@ -3100,6 +3100,11 @@ static int ssl_in_server_key_exchange_parse( mbedtls_ssl_context *ssl,
         {
             mbedtls_platform_random_delay();
 
+            if( rs_ctx == NULL )
```

Comment:
@AndrzejKurek , could you explain this restartable context check?

Reply:
Do we need special return value for this function to prevent skipping ?

Reply:
The thing is that if there's a restartable context set, verification can stop at any given point (because a maximal number of operations has been reached), and then, when it's called the next time, it will continue from this point.
So, now, in context of these double calls - we won't receive the same result with two subsequent calls.
As for the special return value - the checks here are so nested that I wouldn't expect a need to create an additional variable.